modules/tensor_mechanics/include/auxkernels/VectorRankFourAux.h

```diff
@@ -0,0 +1,49 @@
+/****************************************************************/
```

Comment:
I'm not sold on the utility of these classes, and it looks like you're not using them in any test input files, so I can't see examples of how they improve usability (plus that also means that they're not tested). It seems like you can do everything these do using the regular `RankFourAux` or `RankTwoAux` classes. Could you show an example of how you typically would use this?

If you really need this kind of functionality, maybe we should set up an Action that creates the appropriate `RankFourAux` for the index of the tensor for which you want output.

Reply:
The VectorRankTwoAux is used in the tests (see visco_small_strain/gen_maxwell_relax.i for example). Granted, these are mostly for debugging purposes (to check that the time-stepping procedure does its job correctly), but I would still prefer to keep them, in one form or another.

An alternative could be to grant that capability to RankTwoAux/RankFourAux directly instead. I didn't want to touch these auxkernels to limit side effects, but if you believe this would be a better approach, I can change that.

Reply:
I searched just for usage of VectorRankFourAux. Those tests all just have a single tensor listed, though, so there's not much of a benefit there over just using RankFourAux. I assume the use case is that you have a big list of tensors that you're interested in, and you can switch between them by just changing the index (rather than the name of the tensor), is that right?

Reply:
You are correct. If you look at LinearViscoelasticityBase, you'll see that they define a list of RankFourTensor (the stiffnesses of all springs in the spring-dashpot assembly) and one of RankTwoTensor (the internal strains associated to each spring-dashpot pair). These AuxKernels are here so that we can look into these variables (the VectorRankFourAux comes in handy for aging visco-elastic materials, like the concrete model I want to implement or the one Naman wants to write). I can add a test for these if needed.